core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -103,10 +118,12 @@ public CallOptions withCompression(@Nullable String compressorName) {
    * @param deadlineNanoTime the deadline in the clock as per {@link System#nanoTime()}.
    *                         {@code null} for unsetting the deadline.
    */
+  @Deprecated
   public CallOptions withDeadlineNanoTime(@Nullable Long deadlineNanoTime) {
```

Comment:
FWIW, they can probably never go away.

Methods marked deprecated need to point to the correct way to do something in their comments.